test/e2e/tests/commons/sinspqa/sinsp.py

```diff
@@ -257,19 +257,27 @@ def assert_events(expected_events: dict,
         container (docker.Container): A container object to stream logs from.
         timeout (int): The seconds to wait for the events to be asserted
     """
-    for event in expected_events:
-        success = False
-        received_event = None
+    start = datetime.now()
```

Comment:
Idea is:
* make use of timeout
* for each log line, try to match it against all expected events
* every time an event gets matched, remove it from the `expected_events` list
* if we reach the timeout before matching all events, throw an assert failure

Previously, the code was error prone:
* imagine `len(expected_events) == 2` 
* imagine `sinsp.read()` returned 50 lines
* we matched `expected_events[0]`
* but in the remaining read lines there was also `expected_events[1]`
* we did not try to match it against all the remaining lines
* next for iteration, we try to match `expected_events[1]` against a new read of sinsp-example output that **won't** contain the requested event (because we already consumed the log lines where the event was present)